src/server/passive_tran_server.cpp

```diff
@@ -22,6 +22,11 @@
 #include "system_parameter.h"
 #include "thread_manager.hpp"
 
+passive_tran_server::~passive_tran_server ()
+{
+  assert (m_replicator == nullptr);
```

Comment:
I think this assert does not currently hold

Reply:
Is the assert needed?

It seems to me that replicator can just be stopped; since its job is just to keep PTS up to date while running, it loses relevance during shutdown. I may just be stopped whenever.

Reply:
Added code to stop the replication upon server shutdown. Left assert in place.